src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -925,20 +936,60 @@ export default class ResultsViewOncoprint extends React.Component<
         invoke: async () => getUnalteredUids(this.geneticTracks.result!),
     });
 
-    readonly treatmentSelectOptions = remoteData<ISelectOption[]>({
-        await: () => [this.props.store.treatmentsInStudies],
-        invoke: async () =>
-            treatmentsToSelectOptions(
-                this.props.store.treatmentsInStudies.result || []
-            ),
+    readonly genericAssayEntitiesSelectOptionsGroupByGenericAssayType = remoteData<{
+        [genericAssayType: string]: ISelectOption[];
+    }>({
+        await: () => [
+            this.props.store.genericAssayEntitiesGroupByGenericAssayType,
+        ],
+        invoke: async () => {
+            return Promise.resolve(
+                genericAssayEntitiesToSelectOptionsGroupByGenericAssayType(
+                    this.props.store.genericAssayEntitiesGroupByGenericAssayType
+                        .result || {}
```

Comment:
are you sure that the result can be "falsey" ?

Reply:
Oh, It cannot. I should just use this `this.props.store.genericAssayEntitiesGroupByGenericAssayType.result`.